core/plugins/org.polarsys.capella.core.sirius.analysis/src/org/polarsys/capella/core/sirius/analysis/ABServices.java

```diff
@@ -727,6 +727,11 @@ public boolean isValidDndABComponent(Part semanticObjectToDrop, EObject targetCo
     }
 
     context = CsServices.getService().getABTarget((DSemanticDecorator) targetContainerView);
+    final EObject semanticTarget = context; 
+    boolean isSourceAlreadyReployed = semanticObjectToDrop.getDeployingLinks().stream().anyMatch(link -> link.eContainer().equals(semanticTarget));
```

Comment:
The variable `semanticTarget` is useless. Use `context` variable directly.

Reply:
Seems weird I know, but i can't use the `context` variable here since it is not final, and i can't make it final.

Reply:
Yes you can make it final by removing the useless initialization to null.